### PR TITLE
own: basic person search result

### PR DESCRIPTION
**Pull request by limitedmage**
Closes https://github.com/sourcegraph/sourcegraph/issues/48067
Closes https://github.com/sourcegraph/sourcegraph/issues/46949

Returns a basic search result for the owner types (person/team/unknown owner).
Also enables syntax highlighting in the search box for `select:file.owners`

## Test plan

image

## App preview:

- [Web](https://sg-web-jp-onwersearchresult.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.04 kb) | 0.01% (+2.11 kb) | 0.02% (+2.06 kb) | 0.14% (+1) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/jp/onwersearchresult/compare-report.html;tab=live_object) for a full comparison between the commits 541a24f and 9645fa0 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by cbart — APPROVED** [review]
I see nothing wrong here with my limited React/TS knowledge.

File: client/shared/src/components/UserAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { Maybe } from '../graphql-operations'
+
+import styles from './UserAvatar.module.scss'
+
+interface Props {
+    size?: number
+    user: {
+        avatarURL: Maybe<string>
+        displayName: Maybe<string>
+        username?: Maybe<string>
```
[the remark this answers is not in the document]
Comment (limitedmage) [reply client/shared/src/components/UserAvatar.tsx:16]: `Maybe` is from GraphQL and it means the field could be null. `?` is from TypeScript and it means the field could be missing (`undefined`). Yes, JavaScript has both `null` and `undefined` values and they mean very different things :(

**Comment by limitedmage** [comment]
I'll wait until https://github.com/sourcegraph/sourcegraph/pull/47214 lands before so I can update this PR to reflect the schema changes there.

**Review by eseliger — APPROVED** [review]
LGTM, pending the API change! Looks slick :)

File: client/shared/src/search/stream.ts [hunk 2]
```diff
@@ -579,6 +603,10 @@ export function getMatchUrl(match: SearchMatch): string {
             return getCommitMatchUrl(match)
         case 'repo':
             return getRepoMatchUrl(match)
+        case 'unknownOwner':
+        case 'person':
+        case 'team':
+            return '#' // Once we have enough data, link to a user/team page
```
Comment (eseliger) [inline client/shared/src/search/stream.ts:609]: reminder this is still in here 😬

Comment (limitedmage) [reply client/shared/src/search/stream.ts]: This has been updated!

**Pull request merged**